<!-- -*- xml -*- -->
<launch>

  <arg name="marker_size" default="0.100" doc="Size of the ArUco marker used, in meters" />
  <arg name="marker_id" default="530" doc="The ID of the ArUco marker used" />
  <arg name="corner_refinement" default="LINES" />
  <arg name="simulated" default="false" />
  <arg name="namespace_prefix" default="edo" />

  <!-- start camera -->
  <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" /> 
  
  <!-- start robot 
  <include file="$(find customized_edo)/launch/demo.launch" >
    <arg name="simulated" value="$(arg simulated)"/>
  </include>
  -->

  
  <!-- start the robot 
  <include file="$(find customized_edo)/launch/demo.launch">
  </include> -->

  <!-- start ArUco  -->
  <node name="aruco_tracker" pkg="aruco_ros" type="single">
    <remap from="/camera_info" to="/camera/color/camera_info" /> 
    <remap from="/image" to="/camera/color/image_rect_color" />
    <param name="image_is_rectified" value="true"/>
    <param name="marker_size"        value="$(arg marker_size)"/>
    <param name="marker_id"          value="$(arg marker_id)"/>
    <param name="reference_frame"    value="camera_link"/>
    <param name="camera_frame"       value="camera_color_optical_frame"/>
    <param name="marker_frame"       value="marker_frame" />
    <param name="corner_refinement"  value="$(arg corner_refinement)" />
  </node> 
 

  <!-- start easy_handeye -->
  <include file="$(find positionbased_vs)/launch/calibrate.launch" >
  
    <arg name="move_group" value="edo"  doc="the name of move_group for the automatic robot motion with MoveIt!" />
    <arg name="eye_on_hand" value="false" />
    <arg name="tracking_base_frame" value="camera_link" />
    <arg name="tracking_marker_frame" value="marker_frame" />
    <arg name="robot_base_frame" value="edo_base_link" />
    <arg name="robot_effector_frame" value="edo_gripper_link_ee" />

    <arg name="freehand_robot_movement" value="false" />
    <arg name="robot_velocity_scaling" value="0.5" />
    <arg name="robot_acceleration_scaling" value="0.2" />
    <arg name="rotation_delta_degrees" value="15" />
    <arg name="translation_delta_meters" value="0.1" />
  </include>
  
</launch>


